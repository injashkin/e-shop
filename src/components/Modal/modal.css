.modal {
  display: none;
  width: 100vw;
  height: 100vh;
  background-color: rgba(0, 0, 0, 0.4);
  position: fixed;
  top: 0;
  left: 0;
  z-index: 599;
}

.modal__content {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  position: fixed;
  width: 46rem;
  height: 26.5rem;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  background-color: #ffffff;
  z-index: 1000;
  border-radius: 0.5rem;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
}

.modal__img-close {
  position: absolute;
  top: 1.875rem;
  right: 1.875rem;
}

.modal__title {
  font-weight: 500;
  font-size: 40px;
  line-height: 120%;
  text-align: center;
  text-transform: uppercase;
  color: #111111;
  margin: 1.5rem 0 1.25rem;
}

.modal__text {
  font-weight: 300;
  font-size: 20px;
  line-height: 120%;
  text-align: center;
  color: #3F4E65;
}

@media (max-width: 760px) {
  .modal__content {
    width: 18.125rem;
    height: 18.125rem;
  }

  .modal__img-close {
    top: 0.625rem;
    right: 0.625rem;
  }

  .modal__title {
    font-size: 20px;
    margin: 1.25rem 0 0.625rem;
  }

  .modal__text {
    font-size: 12px;
  }
}