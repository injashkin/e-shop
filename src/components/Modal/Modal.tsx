import "./modal.css";
import close from "../../assets/close.svg";
import { ReactNode } from "react";

interface IModal {
  show?: boolean;
  className?: string;
  onClose: (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => void;
  title?: string;
  text?: string;
  children?: ReactNode;
}

const Modal = ({
  show = true,
  title,
  text,
  onClose,
  className,
  children,
}: IModal) => {
  let classes = "";

  if (show) {
    classes = className
      ? `modal modal--show ${className}`
      : "modal modal--show";
  } else classes = className ? `modal ${className}` : "modal";
  //className={`modal${className && `" "${className}`}`}

  return (
    <div className={classes} onClick={(e) => onClose(e)}>
      <div className="modal__content" onClick={(e) => e.stopPropagation()}>
        <img
          className="modal__img-close"
          src={close}
          alt="Close"
          onClick={(e) => onClose(e)}
        />
        <>{children}</>
        <div className="modal__title">{title}</div>
        <p className="modal__text">{text}</p>
      </div>
    </div>
  );
};

export default Modal;
