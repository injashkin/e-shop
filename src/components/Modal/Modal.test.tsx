import { fireEvent, getByText, render, screen } from "@testing-library/react";
import Modal from "./Modal";

describe("Модальное окно", () => {
  test("должен отобразить окно с текстом", () => {
    const component = render(
      <Modal
        title="Спасибо за заказ"
        text="Наш менеджер свяжется с вами в ближайшее время"
        show={true}
        onClose={(e) => e}
      ></Modal>
    );
    expect(component).toMatchSnapshot();
  });

  test("должен закрыть окно при нажатии на крестик", () => {
    const closeModal = vi.fn();

    const component = render(
      <Modal show={true} onClose={(e) => closeModal(e)}>
        <div>test</div>
      </Modal>
    );
    expect(screen.getByText("test")).toBeTruthy();

    fireEvent.click(screen.getByAltText(/close/i));
    expect(closeModal).toHaveBeenCalledTimes(1);
  });

  test("должен закрыть окно при нажатии на крестик", () => {});
});
