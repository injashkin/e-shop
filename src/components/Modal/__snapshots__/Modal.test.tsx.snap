// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Модальное окно > должен отображать окно с текстом 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="modal modal--show"
      >
        <div
          class="modal__content"
        >
          <img
            alt="Close"
            class="modal__img-close"
            src="/src/assets/close.svg"
          />
          <div
            class="modal__title"
          >
            Спасибо за заказ
          </div>
          <p
            class="modal__text"
          >
            Наш менеджер свяжется с вами в ближайшее время
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="modal modal--show"
    >
      <div
        class="modal__content"
      >
        <img
          alt="Close"
          class="modal__img-close"
          src="/src/assets/close.svg"
        />
        <div
          class="modal__title"
        >
          Спасибо за заказ
        </div>
        <p
          class="modal__text"
        >
          Наш менеджер свяжется с вами в ближайшее время
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Модальное окно > должен отобразить окно с текстом 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="modal modal--show"
      >
        <div
          class="modal__content"
        >
          <img
            alt="Close"
            class="modal__img-close"
            src="/src/assets/close.svg"
          />
          <div
            class="modal__title"
          >
            Спасибо за заказ
          </div>
          <p
            class="modal__text"
          >
            Наш менеджер свяжется с вами в ближайшее время
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="modal modal--show"
    >
      <div
        class="modal__content"
      >
        <img
          alt="Close"
          class="modal__img-close"
          src="/src/assets/close.svg"
        />
        <div
          class="modal__title"
        >
          Спасибо за заказ
        </div>
        <p
          class="modal__text"
        >
          Наш менеджер свяжется с вами в ближайшее время
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
